include/swift/ABI/TaskOptions.h

```diff
@@ -78,23 +78,25 @@ class TaskGroupTaskOptionRecord : public TaskOptionRecord {
 
 /// Task option to specify on what executor the task should be executed.
 ///
-/// Not passing this option implies that a "best guess" or good default
-/// executor should be used instead, most often this may mean the global
-/// concurrent executor, or the enclosing actor's executor.
-class ExecutorTaskOptionRecord : public TaskOptionRecord {
-  const SerialExecutorRef Executor;
+/// Not passing this option implies that an inferred (e.g. surrounding actor when we inherit execution context)
+/// or the default executor should be used.
+///
+/// Lack of this option usually means that the global concurrent executor, or the executor of the enclosing actor
+/// will be used.
```

Comment:
Please word-wrap comments at ~80 columns

Reply:
Ok I'll make sure to clang format it all

Reply:
Okey, yeah I wasn't sure if we want to hardcode this assumption here -- you are right that it is guaranteed currently to be a `{0, 0}` basically (SerialExecutorRef), so okey if we want to assume this I removed the function and replaced it with:

```
@available(SwiftStdlib 9999, *)
@usableFromInline
internal func _getGenericSerialExecutor() -> Builtin.Executor {
  // The `SerialExecutorRef` to "default generic executor" is guaranteed
  // to be a zero value;
  //
  // As the runtime relies on this in multiple places,
  // so instead of a runtime call to get this executor ref, we bitcast a "zero"
  // of expected size to the builtin executor type.
  unsafeBitCast((UInt(0), UInt(0)), to: Builtin.Executor.self)
}
```

Is that the right UInt to use here...? In C++ it is:

```
  HeapObject *Identity; // Not necessarily Swift reference-countable
  uintptr_t Implementation;
```